test/gmonoid.lean

```diff
@@ -0,0 +1,30 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import algebra.direct_sum.ring
+import data.fin.tuple.basic
+
+/-! # Tuples `fin n → α` form a graded monoid with `*` as `fin.append`
+
+Defining multiplication as concatenation isn't particularly canonical, so we do not provide
+this in mathlib. We could safely provide this instance on a type alias, but for now we just put
+this in `tests` to verify that this definition is possible. -/
```

Comment:
We have `free_monoid` which is a type alias of `list`. Would you make it an internally graded monoid (by length)?

Reply:
I can certainly PR that separately, but it wouldn't be a test of external gradations